src/TorchSharp/NN/Module.cs

```diff
@@ -251,7 +251,9 @@ private void _toEpilog(ScalarType? dtype, Device device)
                         // In addition, we need the new tensor to be a leaf to accumulate gradients, so if we didn't
                         // disable grad we would need to call .detach() on the moved tensor.
                         using (var d = torch.no_grad())
-                            p = new Parameter(param.to(paramType, device ?? param.device, disposeAfter: true), requiresGrad);
+                            p = new Parameter(
+                                param.to(paramType, device ?? param.device, disposeAfter: true).DetachFromDisposeScope(), requiresGrad)
+                                .DetachFromDisposeScope() as Parameter;
```

Comment:
I may be wrong about this - If I understood the Parameter class correctly, the handle is moved from the Tensor to the Parameter on creation (and the Tensor's handle is set to 0, so we don't care if it's disposed), and therefore we don't need to detach the internal tensor, just the final parameter.

Reply:
Yeah... better safe than sorry.

Reply:
Fair enough :)